rosidl_runtime_py/rosidl_runtime_py/convert.py

```diff
@@ -50,18 +53,22 @@ def message_to_yaml(msg: Any, truncate_length: int = None) -> str:
         __yaml_representer_registered = True
 
     return yaml.dump(
-        message_to_ordereddict(msg, truncate_length=truncate_length),
+        message_to_ordereddict(
+            msg, truncate_length=truncate_length, no_arr=no_arr, no_str=no_str),
         width=sys.maxsize,
     )
 
 
-def message_to_csv(msg: Any, truncate_length: int = None) -> str:
+def message_to_csv(msg: Any, truncate_length: int = None,
+                   no_arr: bool = False, no_str: bool = False) -> str:
     """
     Convert a ROS message to string of comma-separated values.
 
     :param msg: The ROS message to convert.
     :param truncate_length: Truncate values for all message fields to this length.
         This does not truncate the list of message fields.
+    :param no_arr: Exclude array fields of the message
+    :param no_str: Exclude string fields of the message
     :returns: A string of comma-separated values representing the input message.
     """
     def to_string(val):
```

Comment:
@vinnamkim this isn't dealing with e.g. strings in arrays.

Reply:
Fixed.